### ISSUE TITLE
False positives with `Style/DisableCopsWithinSourceCodeDirective`.

### ISSUE THREAD

**Issue by mockdeep** [sds/haml-lint#406]
When upgrading to version 0.46.0, we're seeing a lot of false positives with the `Style/DisableCopsWithinSourceCodeDirective` linter. The following line triggers it for some reason:

```haml
%li.tab-navigation__tab{ **tab_attributes }
```

**Comment by MaxLap** [comment]
I'm learning about a new Cop I didn't know about...

We use the `begin # rubocop:disable Style/RedundantBegin,Lint/RedundantCopDisableDirective` when we encounter tags to add valid indentation to the Ruby code send to Rubocop. (Using a block would cause scoping issues with variables)

I made a PR (#412) to fix this by instead adding a `ensure` to the begin and remove the disables. This should fix your issue.

In the meantime, I can only suggest to disable DisableCopsWithinSourceCodeDirective or to add `AllowedCops` for Style/RedundantBegin and Lint/RedundantCopDisableDirective

**Comment by MaxLap** [comment]
Haml-Lint 0.47.0 has been released which fixes this. This can be closed
